### PR TITLE
vscode-langservers-extracted: expose vscode-eslint, switch to vscodium

### LINKED ISSUE #239425: `vscode-eslint-language-server` missing from `vscode-langservers-extracted`
### Describe the bug
Prior to `vscode-langservers-extracted` being repackaged in this PR https://github.com/NixOS/nixpkgs/pull/234888, a binary called `vscode-eslint-language-server` was exposed. 
This is the binary/lsp that is used by neovim lspconfig by default https://github.com/neovim/nvim-lspconfig/blob/master/lua/lspconfig/server_configurations/eslint.lua#L47

I see in the PR that eslint wasn't in scope because of the need for `vscode-eslint`, but what else can be done?

### Steps To Reproduce
1. use `nixos-unstable`
2. add nodePackages.vscode-langservers-extracted to home.packages
3. switch
4. try to find `vscode-eslint-language-server`

### Expected behavior
`vscode-eslint-language-server` should be an accessible binary on `$PATH`

### Notify maintainers

@Lord-Valen 
@SuperSandro2000 
@malob 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 22.5.0, macOS 13.4`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.15.1`
 - channels(root): `"nixpkgs"`
 - channels(aordonez): `"darwin"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

### PR DESCRIPTION
**Pull request by Lord-Valen**
###### Description of changes

Adds `vscode-eslint-language-server` as was planned in #234888.

Closes #239425

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by malob** [comment]
Result of `nixpkgs-review pr 239650` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vscode-langservers-extracted</li>
  </ul>
</details>

**Comment by malob** [comment]
Result of `nixpkgs-review pr 239650` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vscode-langservers-extracted</li>
  </ul>
</details>

**Review by malob — APPROVED** [review]
Builds fine for me on macOS.

**Review by linsui — APPROVED** [review]

**Comment by linsui** [comment]
Could you please replace vscode with vscodium so that this package becomes free.

```nix
        then "${vscodium}/Applications/VSCodium.app/Contents/Resources/app/extensions"
        else "${vscodium}/lib/vscode/resources/app/extensions";
```

**Comment by malob** [comment]
@linsui, I didn't test making that change, but I did just look at the contents of `${vscodium}/Applications/VSCodium.app/Contents/Resources/app/extensions` and it does seem like all the required files are there, so it looks like this should be possible.

File: pkgs/development/tools/language-servers/vscode-langservers-extracted/default.nix [hunk 1]
```diff
@@ -11,20 +10,15 @@ buildNpmPackage rec {
...
+  npmDepsHash = "sha256-DhajWr+O0zgJALr7I/Nc5GmkOsa9QXfAQpZCaULV47M=";
 
   buildPhase =
     let
+      inherit (vscode-extensions.dbaeumer) vscode-eslint;
       extensions =
         if stdenv.isDarwin
-        then "${vscode}/Applications/Visual\\ Studio\\ Code.app/Contents/Resources/app/extensions"
-        else "${vscode}/lib/vscode/resources/app/extensions";
+        then "${vscodium}/Applications/Visual\\ Studio\\ Code.app/Contents/Resources/app/extensions"
```
Comment (linsui) [inline pkgs/development/tools/language-servers/vscode-langservers-extracted/default.nix:20]:
```suggestion
        then "${vscodium}/Applications/VSCodium.app/Contents/Resources/app/extensions"
```

**Comment by malob** [comment]
Result of `nixpkgs-review pr 239650` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vscode-langservers-extracted</li>
  </ul>
</details>

**Comment by malob** [comment]
Result of `nixpkgs-review pr 239650` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vscode-langservers-extracted</li>
  </ul>
</details>

**Review by linsui — APPROVED** [review]

File: pkgs/development/tools/language-servers/vscode-langservers-extracted/default.nix [hunk 2]
```diff
@@ -1,5 +1,4 @@
-{ lib, stdenv, buildNpmPackage, fetchFromGitHub, vscode }:
-
+{ lib, stdenv, buildNpmPackage, fetchFromGitHub, vscodium, vscode-extensions }:
```
Comment (SuperSandro2000) [inline pkgs/development/tools/language-servers/vscode-langservers-extracted/default.nix:1]:
```suggestion
{ lib, stdenv, buildNpmPackage, fetchFromGitHub, vscodium, vscode-extensions }:

```

File: pkgs/development/tools/language-servers/vscode-langservers-extracted/default.nix [hunk 3]
```diff
@@ -11,20 +10,15 @@ buildNpmPackage rec {
...
-  postPatch = ''
-    # TODO: Add vscode-eslint as a dependency
-    # Eliminate the vscode-eslint bin
-    sed -i '/^\s*"vscode-eslint-language-server":.*bin\//d' package.json package-lock.json
-  '';
+  npmDepsHash = "sha256-DhajWr+O0zgJALr7I/Nc5GmkOsa9QXfAQpZCaULV47M=";
 
   buildPhase =
     let
+      inherit (vscode-extensions.dbaeumer) vscode-eslint;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/language-servers/vscode-langservers-extracted/default.nix:17]:
```suggestion
```

File: pkgs/development/tools/language-servers/vscode-langservers-extracted/default.nix [hunk 4]
```diff
@@ -35,11 +29,13 @@ buildNpmPackage rec {
         --out-dir lib/json-language-server/node/
       npx babel ${extensions}/markdown-language-features/server/dist/node \
         --out-dir lib/markdown-language-server/node/
+      cp -r ${vscode-eslint}/share/vscode/extensions/dbaeumer.vscode-eslint/server/out \
```
Comment (SuperSandro2000) [inline pkgs/development/tools/language-servers/vscode-langservers-extracted/default.nix:32]:
```suggestion
      cp -r ${vscode-extensions.dbaeumer.vscode-eslint}/share/vscode/extensions/dbaeumer.vscode-eslint/server/out \
```

**Pull request merged**